modules/azure-virtual-network-gateway/vars.tf

```diff
@@ -17,14 +17,6 @@ variable "ip_configuration" {
     ])
     error_message = "Valid options for `ip_configuration.private_ip_address_allocation` are \"Static\" or \"Dynamic\"."
   }
-  validation {
-    condition = alltrue([
-      # TODO: change to endswith()
-      # https://github.com/hashicorp/terraform/pull/31220
-      for ip_config in var.ip_configuration : can(regex("GatewaySubnet$", ip_config["subnet_id"]))
-    ])
-    error_message = "It is mandatory that the associated subnet is named \"GatewaySubnet\"."
-  }
```

Comment:
So it's no longer a requirement that `subnet_id` must end with `GatewaySubnet`?

Reply:
it is, but we can't validate this, as `ID` is only generated `post-apply`